rpcserver.go

```diff
@@ -456,6 +460,83 @@ func determineFeePerKw(feeEstimator lnwallet.FeeEstimator, targetConf int32,
 	}
 }
 
+func (r *rpcServer) ListUnspent(ctx context.Context,
+	in *lnrpc.ListUnspentRequest) (*lnrpc.ListUnspentResponse, error) {
+	minConfs := in.Minconfs
+	maxConfs := in.Maxconfs
+	if minConfs >= maxConfs {
+		return nil, fmt.Errorf("Max confirmations must be > min " +
+			"confirmations")
+	}
+	usingMax := maxConfs != math.MaxInt32
+	if usingMax {
+		rpcsLog.Infof("[listunspent] min=%v max=%v", minConfs, maxConfs)
+	} else {
+		rpcsLog.Infof("[listunspent] min=%v", minConfs)
+	}
+
+	utxos1, err := r.server.cc.wallet.ListUnspentWitness(minConfs)
+	if err != nil {
+		return nil, err
+	}
+
+	utxos := make([]*lnwallet.Utxo, 0, len(utxos1))
+
+	if usingMax {
+		utxos2, err := r.server.cc.wallet.ListUnspentWitness(maxConfs)
+		if err != nil {
+			return nil, err
+		}
+		// We want the symmetric difference between the first set (utxos1)
```

Comment:
A better solution would be to modify the result struct to inlcude the number of confirmations: https://github.com/lightningnetwork/lnd/blob/b00e43eec1f19e55f910f7eb527bf7437b4069b4/lnwallet/interface.go#L52

This will also be useful other places, and makes sure we list unspents in one atomic db transaction! 😀

Reply:
Thanks, so, just to check I understood: I mentioned in the first comment in the thread that I wanted to, but wasn't able, to include the number of confs due to that `Utxo` struct not containing it; so you're suggesting here that I do change that result struct. I was a bit wary of messing with more "internal" code for this PR, but I guess there's no possibility of that causing an issue elsewhere.

Reply:
Oh sorry, didn't see that comment 😛 But yeah, I think it should be fairly trivial and non-disruptive, as you have the number of confirmations available as part of the returned result here: https://github.com/lightningnetwork/lnd/blob/b00e43eec1f19e55f910f7eb527bf7437b4069b4/lnwallet/btcwallet/btcwallet.go#L305

Reply:
The only think remaining is to add the same information to the `Utxo` struct, correct 👍

Reply:
Fixed in 60ae2afc236c6eb3d1095cca9dfed5d9760a23b4